### DISCUSSION TITLE
Size argument missing in TransformerConv.forward(...)

### DISCUSSION THREAD

**Discussion by aabbas90** [pyg-team/pytorch_geometric#2272] [Q&A]
Hi,

I am trying to use [`TransformerConv`](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/conv/transformer_conv.html#TransformerConv.forward) layer on bipartite graph **G** = ((**V_l**, **V_r**), **E**). The goal is to send information from right vertex set **V_r** and edges **E** to **V_l** and then vice-versa. The code is:

```

left_updater= TransformerConv((num_node_feat, num_node_feat), num_node_feat, edge_dim = num_edge_feat, flow="source_to_target")
right_updater = TransformerConv((num_node_feat, num_node_feat), num_node_feat, edge_dim = num_edge_feat, flow="target_to_source")

left_updated_features = left_updater((left_features, right_features), edge_index_left_right, edge_features)
right_updated_features = right_updater((left_features, right_features), edge_index_left_right, edge_features)

```
However, I am getting the following error:

 >  File "~/anaconda3/envs/learnDA/lib/python3.7/site-packages/torch_geometric/nn/conv/message_passing.py", line 127, in __lift__
>     return src.index_select(self.node_dim, index)
> IndexError: index out of range in self

I guess it might be because I am not passing the `size` argument to give the size of nodes in **V_l, V_r** because the `TransformerConv.forward(...)` does not accept it as an argument. Would be happy to get your feedback on how to use TransformerConv on bipartite graphs. Thanks!

**Comment by rusty1s** [comment]
According to our tests, `TransformerConv` should work on bipartite graphs just fine, e.g.,
```python
conv = TransformerConv((8, 16), 32)
conv((x1, x2), edge_index)
```
I guess this is induced by the `flow="target_to_source"` modification. Note that in this case, you need to flip the `(left_features, right_features)` formulation, i.e.
```python
right_updated_features = right_updater((right_features, left_features), edge_index_left_right, edge_features)
```
```
